### ISSUE TITLE
[BUG] Your next.js starter app has issues

### ISSUE THREAD

**Issue by ahhshm** [refinedev/refine#3259]
### Describe the bug

I found a couple of issues with your next.js starter app.
```ts
// pages/[[...refine]].tsx
export const getServerSideProps: GetServerSideProps = async (context) => {
  // const { resource, action, id } = handleRefineParams(context.params?.refine);

  const { isAuthenticated, ...props } = await checkAuthentication(
    authProvider,
    context
  );

  const i18nProps = await serverSideTranslations(context.locale ?? "en", [
    "common",
  ]);

  if (!isAuthenticated) {
    return { props: { ...props, ...i18nProps } };
  }

  return {
    props: {
      ...(await serverSideTranslations(context.locale ?? "en", ["common"])),
    },
  };
};
```
1. `i18nProps` is already calculated and stored in a variable so don't need to run `await serverSideTranslations(context.locale ?? "en", ["common"])` again. It should be:
```ts
  return {
    props: {
      ...i18nProps,
    },
  };
```
2. Here `props` could be any valid return value of `getServerSideProps`. 
```ts
  if (!isAuthenticated) {
    return { props: { ...props, ...i18nProps } };
  }
```
For example it could be a redirect, but your are spreading it into the `props` ? I think it should be something like this:
```ts
  if (!isAuthenticated) {
    return {
      ...props,
      // @ts-ignore
      props: { ...(props?.props ? props.props : {}), ...i18nProps },
    };
  }
```
Now redirects work. I had to add `@ts-ignore` because otherwise I would get a wired typescript error that I couldn't resolve.

If I know where is your next.js starter app code, I would update it myself.

### Steps To Reproduce

x

### Expected behavior

x

**Comment by aliemir** [comment]
Hey @ahhshm, thank you for the explanation 🙏 Looks like you figured out a better approach. We have a `i18n-nextjs` example located at `examples/i18n-nextjs`. If you want you can start by updating this example and checking if there's any issues when we apply those changes. 

For the starter app, we use [`superplate`](https://github.com/pankod/superplate) to create projects by users choices. You can find the code for the Refine + Next.js apps and its supported plugins in [`superplate-core-plugins`](https://github.com/pankod/superplate-core-plugins), the source you're looking for is inside `refine-nextjs`. 

Basically, we're creating files depending on the user's choices with EJS. If you want to give it a try, we'll be happy to accept your PRs 🎉 🎉 

Let me know if you need any help if you decide to create a PR for your changes 🚀
